### DISCUSSION TITLE
How to change the time zone of OPA Built-in Function - Time.clock()

### DISCUSSION THREAD

**Discussion by JinXiu2023** [open-policy-agent/community#425] [OPA and Rego] [answered]
Hello, I got a problem when I use the function -time.clock(). I want to get the time in Asia/Shanghai, but I always get the time 8 hours later than Shanghai's time even though I have changed the POD's time zone. I need your help and thank you anyway.

**Comment by charlieegan3** [comment, marked as answer]
Hey, how about something like this: https://play.openpolicyagent.org/p/LvzRDk66Rg

```rego
t := time.clock([time.now_ns(), "Asia/Shanghai"])
```

**Reply by mariano-dm** [reply]
it returns {}

**Reply by anderseknert** [reply]
Huh, it does indeed. Some recent regression? 🤔

**Reply by charlieegan3** [reply]
Looks like it might be a playground issue.

```
$ opa eval 'time.clock([time.now_ns(), "Asia/Shanghai"])'
{
  "result": [
    {
      "expressions": [
        {
          "value": [
            17,
            13,
            12
          ],
          "text": "time.clock([time.now_ns(), \"Asia/Shanghai\"])",
          "location": {
            "row": 1,
            "col": 1
          }
        }
      ]
    }
  ]
}
$ opa version
Version: 0.53.1
```

**Reply by charlieegan3** [reply]
If you turn on errors you can see `1 error occurred: policy.rego:3: eval_builtin_error: time.clock: unknown time zone Asia/Shanghai`. Just looks like the playground container is missing some TZ data.

**Reply by anderseknert** [reply]
💡

**Reply by charlieegan3** [reply]
https://github.com/open-policy-agent/opa/pull/6038
